src/calls.c

```diff
@@ -1388,7 +1388,7 @@ Obj FuncNARG_FUNC (
     Obj                 func )
 {
     if ( TNUM_OBJ(func) == T_FUNCTION ) {
-        return INTOBJ_INT( NARG_FUNC(func) );
+        return ObjInt_Int(NARG_FUNC(func));
```

Comment:
That really can't matter:
```C
EXPORT_INLINE Int NARG_FUNC(Obj func)
{
    return INT_INTOBJ(CONST_FUNC(func)->nargs);
}
```
If you are deeply concerned about it, perhaps this instead?
```suggestion
        return CONST_FUNC(func)->nargs;
```

Reply:
As a general argument, I'd prefer that we use `ObjInt_Int` (and friends) except in places where it really matters -- there is an issue of philosophy here. I increasingly feel we should move towards using methods which are guaranteed not to lead to corrupted / broken states, except in a small number of places where we can carefully verify correctness.

Reply:
I have a somewhat differing view: in many places, I'd agree that `ObjInt_Int` is better (including many covered by this PR). But I feel that sometimes it just obscures what is happening. And this is a prime example. It feels like we are building an onion out of abstraction layers here, and I'd feel much better if we just cut them away and wrote code that avoided pointless back and forth conversions altogether -- which is precisely what the change I suggested here does.

Reply:
I agree with all of this PR *except* for the function body related changes, where I'd prefer the alternative changes I suggested, or no change at all.